### PR TITLE
Doing refactor of CalendarButton and adding historical dashboard

### PR DESCRIPTION
**Pull request by Faustrox**
#### Screenshots: 
Screenshot_2020-09-14-16-58-51-655_com optic covdr
photo5107252801628645663
Screenshot_2020-09-11-17-32-49-496_com optic covdr

### CODE REVIEW

File: app/components/DR/CasesStatistics.js [hunk 1]
```diff
@@ -16,11 +19,17 @@ class CasesStatistics extends React.Component {
       deaths: 0,
       recovered: 0,
       current: 0,
+      date: undefined,
```
Comment (D-Rosa99) [inline app/components/DR/CasesStatistics.js]: Not a good idea to use undefined as default value, always use the same type of value will should get the state

File: app/locales/es.json [hunk 2]
```diff
@@ -9,6 +9,10 @@
     "done": "Hecho",
     "ok": "Ok"
   },
+  "dashboard": {
+    "error_title": "Un error ha ocurrido",
+    "error_message": "No se ha encontrado los datos "
```
Comment (D-Rosa99) [inline app/locales/es.json]: I think this message is not user friendly, it should be some like this, same in english, if you want can make a group discussion about it  
```suggestion
    "error_title": "Oops!, se ha producido un error ",
    "error_message": "Lo sentimos!, no tenemos la informacion de la fecha especificada en nuestra base de datos, por favor pruebe con otra fecha "
```

**Review by D-Rosa99 — CHANGES_REQUESTED** [review]

**Comment by D-Rosa99** [comment]
This information is from 12/09/2020 (DAY BEFORE) but the calendar component shows the current day whose information is not yet 

image

**Comment by D-Rosa99** [comment]
When the error message appeared and I hit `ok` and want to switch from English to Spanish or vice versa the dashboard does not change 

image

File: app/components/DR/CasesStatistics.js [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
-import { View } from 'react-native';
+import { Alert, View } from 'react-native';
 import { heightPercentageToDP as hp } from 'react-native-responsive-screen';
 
+import CalendarButton from '../../components/DR/CalendarButton';
 import { getAllCases } from '../../services/DR/getAllCases.js';
 import styles from '../../views/DR/HomeScreen/style';
 import DashboardCards from './DashboardCards.js';
 
+const miniumDate = '2020-07-17';
```
Comment (D-Rosa99) [inline app/components/DR/CasesStatistics.js]: if you only use this information once in the component, it is not necessary to have it in a variable, please delete it

File: app/components/DR/CasesStatistics.js [hunk 4]
```diff
@@ -102,8 +103,20 @@ class CasesStatistics extends React.Component {
...
+            style={{ backgroundColor: '#FFF' }}
+            date={
+              date
+                ? moment(date, 'YYYY-MM-DD').format('DD-MM-YYYY')
+                : moment(new Date()).format('DD-MM-YYYY')
+            }
+            onChange={date => {
+              this.getCases(moment(date, 'DD-MM-YYYY').format('YYYY-MM-DD'));
+            }}
+            minDate={moment(miniumDate)}
```
Comment (D-Rosa99) [inline app/components/DR/CasesStatistics.js]: I think moment is not necessary here

```suggestion
            minDate='2020-07-17'
```

File: app/services/DR/getAllCases.js [hunk 5]
```diff
@@ -3,14 +3,14 @@ import { AUTHORITY_CASES } from '../../constants/storage';
 import fetch from '../../helpers/Fetch';
 import { GetStoreData, SetStoreData } from '../../helpers/General';
 
-export function getAllCases() {
+export function getAllCases(date) {
   const dataSaved = GetStoreData(AUTHORITY_CASES, false);
```
Comment (D-Rosa99) [inline app/services/DR/getAllCases.js]: you've removed **dataSaved** from the catch and we've been using it.

Won't we use it anymore? if that the case please delete the variable

File: app/components/DR/CasesStatistics.js [hunk 6]
```diff
@@ -102,8 +107,22 @@ class CasesStatistics extends React.Component {
...
+              style={{ backgroundColor: '#FFF' }}
+              date={
+                date
+                  ? moment(date, 'YYYY-MM-DD').format('DD-MM-YYYY')
+                  : moment(new Date()).format('DD-MM-YYYY')
+              }
+              onChange={date => {
+                this.getCases(moment(date, 'DD-MM-YYYY').format('YYYY-MM-DD'));
+              }}
+              minDate={'2020-07-17'}
```
Comment (D-Rosa99) [inline app/components/DR/CasesStatistics.js]: You put the minDate in an incorrect format, the format should be **DD-MM-YYYY** and the brackets are not necessary

```suggestion
              minDate='17-07-2020'
```

image

**Review by D-Rosa99 — CHANGES_REQUESTED** [review]
When I put **03-09-2020** and others the dashboard display an error message, but this date exist in the database, please check out.

I added all the data from 17-07-2020 until current the current day, so there's not should be errors like that with old dates

File: app/components/DR/CasesStatistics.js [hunk 7]
```diff
@@ -102,8 +107,22 @@ class CasesStatistics extends React.Component {
               styles.dateSubtitle,
               { alignSelf: 'center', textAlign: 'center' },
             ]}>
-            {this.getUpdateDate()}
+            {t('label.date_dashboard_label')}
           </Text>
+          <View style={{ alignSelf: 'center' }}>
```
Comment (D-Rosa99) [inline app/components/DR/CasesStatistics.js]: it's not necessary to add this `view` to center the calendar, you can add the style in the component, something like this

```
<CalendarButton
              style={{ backgroundColor: '#FFF', alignSelf: 'center' }}
```

**Review by D-Rosa99 — CHANGES_REQUESTED** [review]

File: app/components/DR/CasesStatistics.js (hunk 6 above)
Comment (Faustrox) [reply app/components/DR/CasesStatistics.js:123]: Nop, is correct but the bracket can be deleted

File: app/components/DR/CalendarButton/index.js [hunk 8]
```diff
@@ -27,7 +32,7 @@ export default function CalendarButton({ onChange, date, minDate = '' }) {
       }
       confirmBtnText={t('common.ok')}
       cancelBtnText={t('label.cancel')}
-      style={{ width: wp('40%'), backgroundColor: '#EFF4F9' }}
+      style={({ width: wp('40%'), backgroundColor: '#EFF4F9' }, style)}
```
Comment (RutsSantos) [inline app/components/DR/CalendarButton/index.js:35]: I think you should use an array of styles

File: app/locales/es.json [hunk 9]
```diff
@@ -9,6 +9,10 @@
     "done": "Hecho",
     "ok": "Ok"
   },
+  "dashboard": {
+    "error_title": "Oops!, se ha producido un error",
+    "error_message": "Lo sentimos!, no tenemos la informacion de la fecha especificada en nuestra base de datos, por favor pruebe con otra fecha"
```
Comment (RutsSantos) [inline app/locales/es.json]: you have a typo. It should say **información**

**Review by RutsSantos — APPROVED** [review]

File: app/components/DR/CalendarButton/index.js (hunk 8 above)
Comment (D-Rosa99) [reply app/components/DR/CalendarButton/index.js:35]: this comment has not been resolved yet

Comment (Faustrox) [reply app/components/DR/CalendarButton/index.js:35]: Done

**Review by D-Rosa99 — CHANGES_REQUESTED** [review]
the design of the calendar button does not look quite good.

visually, how users will know that's a button??
if you want you can talk to @RutsSantos to give you some tips about it

**Review by JoseFcoEstrella — APPROVED** [review]

**Review by JoseFcoEstrella — COMMENTED** [review]
I don't like the style of the button, but maybe the letters would look better if they were blue (like a link), instead of having a white background...

**Review by D-Rosa99 — APPROVED** [review]

**Pull request merged**